airbyte-integrations/connectors/source-snowflake/unit_tests/integration/test_discover.py

```diff
@@ -0,0 +1,226 @@
+def _discover(
+        config_builder: ConfigBuilder,
+        sync_mode: SyncMode,
+        state: Optional[List[AirbyteStateMessage]] = None,
+        expecting_exception: bool = False
+) -> EntrypointOutput:
+        catalog = _catalog(sync_mode)
+        config = config_builder.build()
+        return discover(_source(catalog, config, state), config, expecting_exception)
+
+@mock.patch("source_snowflake.streams.snowflake_parent_stream.uuid.uuid4", return_value=_REQUESTID)
+@mock.patch("source_snowflake.source.SnowflakeJwtAuthenticator")
+class DiscoverTest(TestCase):
+
+    @classmethod
+    def _a_table_record(cls):
+        return a_snowflake_response("check_connection", JsonPath("$.'data'.[*]"), id_path=JsonPath("$.[1]"))
+
+    @HttpMocker()
+    def test_discover_no_pagination(self,  uuid_mock, mock_auth, http_mocker: HttpMocker)->None:
+            
+        
+        _given_get_timezone(http_mocker)
+       
+        for i in ["TEST_TABLE_2","TEST_TABLE_1"]:
+            _given_table_with_primary_keys(http_mocker, i)
+            _given_read_schema(http_mocker,i)
+       
+
+        http_mocker.post(
+        table_request().with_show_catalog().with_requestID(_REQUESTID).build(),
+        snowflake_response("check_connection", JsonPath("$.'data'"))
+        .with_record(
+            self._a_table_record().with_id("TEST_TABLE_1")
+        )
+        .with_record(
+            self._a_table_record().with_id("TEST_TABLE_2"))
+        .build()
+        )
+        
+        
+        output= _discover(_config(), sync_mode=SyncMode.full_refresh)
+        assert len(output.catalog.catalog.streams)
+    
+    @expectedFailure # Not Implemented in the source
+    @HttpMocker()
+    def test_discover_pagination(self,  uuid_mock, mock_auth, http_mocker: HttpMocker)->None:
+
+        _given_get_timezone(http_mocker)
+        for i in ["TEST_TABLE_1","TEST_TABLE_2","TEST_TABLE_3"]:
+            _given_table_with_primary_keys(http_mocker, i)
+            _given_read_schema(http_mocker,i )
+       
+
+        http_mocker.post(
+        table_request().with_show_catalog().with_requestID(_REQUESTID).build(),
+        snowflake_response("check_connection", JsonPath("$.'data'"))
+        .with_record(
+            self._a_table_record().with_id("TEST_TABLE_1")
+        )
+        .with_record(
+            self._a_table_record().with_id("TEST_TABLE_2"))
+        .with_pagination()
+        .with_handle(_HANDLE)
+        .build()
+        )
+
+        http_mocker.get(
+            table_request().with_partition(1).with_handle(_HANDLE).build(is_get=True),
+            snowflake_response("check_connection", JsonPath("$.'data'"))
+            .with_record(
+                self._a_table_record().with_id("TEST_TABLE_3")
+            )
+            .with_pagination()
+            .with_handle(_HANDLE)
+            .build()
+        )
+        output= _discover(_config(), sync_mode=SyncMode.full_refresh)
+        assert len(output.catalog.catalog.streams)==3
```

Comment:
I think this test will not pass as-is. It is already failing but it will still fail once the code handles the partition. The reason is that the [SnowflakePaginationStrategy](https://github.com/artefactory/airbyte/blob/68825ae9ee605512337f7423badc95de2319fd53/airbyte-integrations/connectors/source-snowflake/unit_tests/integration/pagination.py) returns three partitions and we only handle two here. Should we update the `SnowflakePaginationStrategy` to return only two partitions?

Reply:
other test are based on 3 partitions. I updated the test.